src/rtracer.js

```diff
@@ -177,12 +208,21 @@ const hapiPlugin = ({
         requestId = request.headers[headerName.toLowerCase()]
       }
       requestId = requestId || requestIdFactory()
+
       als.enterWith(requestId)
       wrapHttpEmitters(request.raw.req, request.raw.res)
 
       return h.continue
     })
 
+    server.ext('onPreResponse', async (request, h) => {
```

Comment:
It's better to register this hook conditionally, i.e. only when `echoHeader` is `true`.